prospero-cli/src/main/java/org/wildfly/prospero/cli/commands/InstallCommand.java

```diff
@@ -0,0 +1,142 @@
+import org.wildfly.prospero.api.WellKnownFeaturePacks;
+import org.wildfly.prospero.api.exceptions.OperationException;
+import org.wildfly.prospero.cli.ActionFactory;
+import org.wildfly.prospero.cli.CliMain;
+import org.wildfly.prospero.cli.Messages;
+import org.wildfly.prospero.cli.ReturnCodes;
+import org.wildfly.prospero.wfchannel.MavenSessionManager;
+import picocli.CommandLine;
+
+@CommandLine.Command(
+        name = "install",
+        description = "Installs a new application server instance.",
+        sortOptions = false
+)
+public class InstallCommand extends AbstractCommand {
+
+    @CommandLine.Option(
+            names = "--dir",
+            required = true,
+            description = "Target directory where the application server is going to be provisioned."
+    )
+    Path directory;
+
+    @CommandLine.ArgGroup(exclusive = true, multiplicity = "1")
+    FeaturePackOrDefinition featurePackOrDefinition;
+
+    @CommandLine.Option(
+            names = "--channel",
+            description = "Channel file URL."
+    )
+    Optional<String> channel;
+
+    @CommandLine.Option(
+            names = "--provision-config",
+            description = "Provisioning configuration file path. This is special JSON configuration file that cotnains list of channel file references and list of remote Maven repositories."
+    )
+    Optional<Path> channelFile;
+
+    @CommandLine.Option(
+            names = "--channel-repo",
+            description = "URL of a remote Maven repository that contains artifacts required to build an application container."
+    )
+    List<URL> channelRepositories;
+
+    @CommandLine.Option(
+            names = "--local-repo",
+            // TODO: Fix description.
+            description = "Path to a local Maven repository."
+    )
+    Optional<Path> localRepo;
+
+    @CommandLine.Option(
+            names = "--offline",
+            description = "Perform installation from local Maven repository only. Offline installation requires --local-repo to be configured."
+    )
+    boolean offline;
+
+    @CommandLine.Option(names = {"-h", "--help"}, usageHelp = true)
+    boolean help;
+
+    static class FeaturePackOrDefinition {
+        @CommandLine.Option(
+                names = "--fpl",
+                required = true,
+                // TODO: Check what the correct notations of fpl are.
+                description = "Feature pack location. This can be a well known shortcut like \"eap-8.0\" or \"wildfly\", " +
+                        "or a fully qualified feature pack location like " +
+                        "\"wildfly-core@maven(org.jboss.universe:community-universe):current\"."
+        )
+        Optional<String> fpl;
+
+        @CommandLine.Option(
+                names = "--definition",
+                required = true,
+                // TODO: Fix description.
+                description = "Definition file path. Not sure what this is."
```

Comment:
That's a Galleon provisioning XML definition (eg. .galleon/provisioning.xml). It's a way to provision multiple FPs, exclude layers etc

Reply:
OK, would "Galleon provisioning XML definition file" be OK description?

Is the definition file a sufficient input to provision a server?

Reply:
> OK, would "Galleon provisioning XML definition file" be OK description?
Yes, sounds good.

> Is the definition file a sufficient input to provision a server?
It requires the user to provide the channel and repository configuration as well - eg. through `provision-config` parameter.

Reply:
OK, makes sense.